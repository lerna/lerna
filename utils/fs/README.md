# `@lerna/fs`

> description TODO

## Usage

TODO

