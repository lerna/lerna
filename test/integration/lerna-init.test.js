import fs from "fs-promise";
import path from "path";
import execa from "execa";
import readPkg from "read-pkg";
import loadJsonFile from "load-json-file";
import initDirName from "../helpers/initDirName";
import initExternalFixture from "../helpers/initExternalFixture";
import replaceLernaVersion from "../helpers/replaceLernaVersion";

expect.addSnapshotSerializer(replaceLernaVersion);

const ROOTDIR = path.resolve(__dirname, "../..");
const PACKAGE = readPkg.sync(ROOTDIR);
const LERNA = path.join(ROOTDIR, PACKAGE.bin.lerna);

const initEmptyDir = () =>
  initDirName("InitCommand/empty").then((dir) => {
    return fs.ensureDir(dir).then(() => dir);
  });

const initFixture = (name) =>
  initExternalFixture(`InitCommand/${name}`);

const parsePackageJson = (cwd) =>
  readPkg(path.join(cwd, "package.json"), { normalize: false });

const parseLernaJson = (cwd) =>
  loadJsonFile(path.join(cwd, "lerna.json"));

const loadMetaData = (cwd) => Promise.all([
  parsePackageJson(cwd),
  parseLernaJson(cwd),
]);

describe("lerna init", () => {
  test.concurrent("initializes empty directory", () => initEmptyDir().then((cwd) => {
    return execa(LERNA, ["init"], { cwd }).then((result) => {
      expect(result.stdout).toMatchSnapshot("stdout: empty directory");

      return loadMetaData(cwd).then(([packageJson, lernaJson]) => {
        expect(packageJson).toMatchSnapshot("package.json: empty directory");
        expect(lernaJson).toMatchSnapshot("lerna.json: empty directory");
      });
    });
  }));

  test.concurrent("updates existing metadata", () => initFixture("updates").then((cwd) => {
    return execa(LERNA, ["init", "--exact"], { cwd }).then((result) => {
      expect(result.stdout).toMatchSnapshot("stdout: updates");

      return loadMetaData(cwd).then(([packageJson, lernaJson]) => {
        expect(packageJson).toMatchSnapshot("package.json: updates");
        expect(lernaJson).toMatchSnapshot("lerna.json: updates");
      });
    });
  }));

  test.concurrent("removes VERSION file", () => initFixture("has-version").then((cwd) => {
    return execa(LERNA, ["init"], { cwd }).then((result) => {
      expect(result.stdout).toMatchSnapshot("stdout: has-version");
    });
  }));
});
