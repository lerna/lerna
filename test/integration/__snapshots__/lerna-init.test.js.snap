// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lerna.json: empty directory 1`] = `
Object {
  "lerna": "__TEST_VERSION__",
  "packages": Array [
    "packages/*",
  ],
  "version": "0.0.0",
}
`;

exports[`lerna.json: updates 1`] = `
Object {
  "commands": Object {
    "bootstrap": Object {
      "hoist": true,
    },
    "init": Object {
      "exact": true,
    },
  },
  "lerna": "__TEST_VERSION__",
  "packages": Array [
    "packages/*",
  ],
  "version": "1.0.0",
}
`;

exports[`package.json: empty directory 1`] = `
Object {
  "devDependencies": Object {
    "lerna": "^__TEST_VERSION__",
  },
}
`;

exports[`package.json: updates 1`] = `
Object {
  "devDependencies": Object {
    "lerna": "__TEST_VERSION__",
  },
  "name": "updates",
}
`;

exports[`stdout: empty directory 1`] = `
"Lerna __TEST_VERSION__
Initializing Git repository.
Creating package.json.
Creating lerna.json.
Successfully initialized Lerna files"
`;

exports[`stdout: has-version 1`] = `
"Lerna __TEST_VERSION__
Creating package.json.
Creating lerna.json.
Removing old VERSION file.
Successfully initialized Lerna files"
`;

exports[`stdout: updates 1`] = `
"Lerna __TEST_VERSION__
Updating package.json.
Updating lerna.json.
Successfully initialized Lerna files"
`;
